packages/apps/reputation-oracle/server/src/modules/user/user.entity.ts

```diff
@@ -37,4 +38,7 @@ export class UserEntity extends BaseEntity implements IUser {
 
   @OneToOne(() => KycEntity, (kyc) => kyc.user)
   public kyc?: KycEntity;
+
+  @OneToOne(() => SiteKeyEntity, (siteKey) => siteKey.user)
+  public siteKey?: SiteKeyEntity;
```

Comment:
I think yes, this will allow sampling of this field along with the rest of the user data. As far as I remember, otherwise typeorm will not fetch data from join tables and will simply return the user.

Reply:
1. When keep just property name and remove OneToOne relation `public kyc?: KycEntity;` from User Entity.
EntityPropertyNotFoundError: Property "kyc" was not found in "UserEntity". Make sure your query is correct.

2. When keep just property name and remove JoinColumn and OneToOne relation `public user: UserEntity;` from Kyc Entity.
TypeError: Cannot read properties of undefined (reading 'joinColumns')

3. When move JoinColumn from Kyc Entity to User Entity
QueryFailedError: column UserEntity.kyc_id does not exist

Reply:
Sergey Dzeranov
> Then leave it as is.